File: ui/client/src/types/route.ts
export interface RouteListItem {
  id: string;
  name: string;
  activityType: string;
  routeCategory: string;
  status: string;
  distanceKm: number;
  elevationGainM: number;
  estimatedTimeSeconds: number;
  tags: string | null;
  routingProfile: string;
  createdAt: string;
  updatedAt: string;
}

export interface RouteDetail {
  id: string;
  name: string;
  description: string | null;
  activityType: string;
  routeCategory: string;
  status: string;
  distanceKm: number;
  elevationGainM: number;
  elevationLossM: number;
  maxElevationM: number;
  minElevationM: number;
  estimatedTimeSeconds: number;
  tags: string | null;
  routingProfile: string;
  sourceActivityId: string | null;
  sourceFileName: string | null;
  createdAt: string;
  updatedAt: string;
  points: number[][] | null;
  waypoints: RouteWaypoint[] | null;
  pois: RoutePoi[] | null;
  profile: unknown | null;
}

export interface RouteWaypoint {
  id: string;
  lat: number;
  lon: number;
  order: number;
}

export interface RoutePoi {
  id: string;
  type: PoiType;
  name: string;
  lat: number;
  lon: number;
  notes?: string;
}

export type PoiType = 'water' | 'parking' | 'refuge' | 'summit' | 'viewpoint' | 'danger' | 'food' | 'camping' | 'custom';

export type RouteCategory = 'loop' | 'out-and-back' | 'traverse' | 'point-to-point';

export type RouteStatus = 'draft' | 'published';

export type RoutingProfile = 'manual' | 'hiking' | 'trail' | 'cycling' | 'road';

export interface RouteCreateRequest {
  name?: string;
  activityType?: string;
  sourceActivityId?: string;
}

export interface RouteUpdateRequest {
  name: string;
  description?: string;
  activityType: string;
  routeCategory: string;
  tags?: string;
  routingProfile: string;
  status: string;
  points?: number[][];
  waypoints?: RouteWaypoint[];
  pois?: RoutePoi[];
}

export interface RouteAutoSaveRequest {
  points?: number[][];
  waypoints?: RouteWaypoint[];
  pois?: RoutePoi[];
}

File: ui/api/Entities/Route.cs
namespace GpxAnalyzer.Api.Entities;

public class Route
{
    public Guid Id { get; set; }
    public string Name { get; set; } = "";
    public string? Description { get; set; }
    public string ActivityType { get; set; } = "trail";
    public string RouteCategory { get; set; } = "";
    public string Status { get; set; } = "draft";
    public string? PointsJson { get; set; }
    public string? WaypointsJson { get; set; }
    public string? PoisJson { get; set; }
    public string? ProfileJson { get; set; }
    public double DistanceKm { get; set; }
    public double ElevationGainM { get; set; }
    public double ElevationLossM { get; set; }
    public double MaxElevationM { get; set; }
    public double MinElevationM { get; set; }
    public double EstimatedTimeSeconds { get; set; }
    public string? Tags { get; set; }
    public string RoutingProfile { get; set; } = "manual";
    public Guid? SourceActivityId { get; set; }
    public string? SourceFileName { get; set; }
    public string Language { get; set; } = "en";
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
}
